cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_utils.h>
+#include <gtest/gtest.h>
+#include <sys/stat.h>
+#include <test_utils.h>
+#include <treelite/c_api.h>
+#include <treelite/c_api_runtime.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include "decisiontree/decisiontree_impl.h"
+#include "ml_utils.h"
+#include "randomforest/randomforest.hpp"
+
+/** check for libc call errors and assert accordingly */
+#define LIBCCALL_CHECK(call)                                    \
```

Comment:
This looks again like a misnomer, as libc functions don't always return an error code. 

Feel free to drop it, and just use `ASSERT()`, as it is only used twice.